src/marqo/tensor_search/throttling/check_and_increment.lua

```diff
@@ -0,0 +1,36 @@
+local set_key = KEYS[1]
+
+local thread_name = ARGV[1]
+local thread_limit = tonumber(ARGV[2])
+local expire_time = tonumber(ARGV[3])
+
+-- Get expiry time
+local now = redis.call('time')[1]
+local expiry_threshold = now - expire_time
+
+-- Expire items in sorted set past the threshhold
+redis.call('zremrangebyscore', set_key, '-inf', expiry_threshold)

```

Comment:
non blocking this PR, perhaps add a comment: 

consider expiring keys as part of the decrement operation, so that it isnt blocking?

Reply:
great idea actually! don't know why i didnt think of that. will put that functionality in next release

Reply:
Can you create an issue for this?